sklearn/base.py

```diff
@@ -328,6 +332,21 @@ def _get_tags(self):
         return tags
 
 
+class DataArrayMixin:
+    """Provide NamedArray output"""
+    _return_dataarray_ = False
+
+    def _check_dataarray(self, X):
+        """Returns a DataArray with the feature names attached, if needed.
+
+        It only attaches the features names to the data if `return_dataarray`
+        is set to `True`."""
+        if self._return_dataarray_:
```

Comment:
I don't understand this mechanism. How does this work currently? this is only set in pipeline? Why?
To not show the dataarray (or rather NamedArray) to users?
And doesn't that need to be set in FeatureUnion and ColumnTransformer as well?
And how does it propagate through meta-estimators?